testsuite/features/core_proxy_branch_network.feature

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2018 SUSE LLC
+# Licensed under the terms of the MIT license.
+#
+# The scenarios in this feature are skipped if there is no proxy
+# ($proxy is nil) or if there is no private network ($private_net is false)
+
+Feature: Setup SUSE Manager for Retail branch network
+  In order to deploy SUSE Manager for Retail solution
+  As the system administrator
+  I want to prepare the branch network
+
+@proxy
+@private_net
+  Scenario: Install or update branch network formulas on the server
+    When I manually install the "branch-network" formula on the server
+    And I manually install the "dhcpd" formula on the server
+    And I manually install the "bind" formula on the server
+    And I wait for "16" seconds
+
+@proxy
+@private_net
+  Scenario: Enable the branch network formulas on the branch server
+    Given I am on the Systems overview page of this "proxy"
+    When I follow "Formulas" in the content area
+    And I check the "branch-network" formula
+    And I check the "dhcpd" formula
+    And I check the "bind" formula
+    And I click on "Save"
+    Then the "branch-network" formula should be checked
+    And the "dhcpd" formula should be checked
+    And the "bind" formula should be checked
+
+@proxy
+@private_net
+  Scenario: Parametrize the branch network
+    Given I am on the Systems overview page of this "proxy"
+    When I follow "Formulas" in the content area
+    And I follow first "Branch Network" in the content area
+    And I enter "eth1" in NIC field
+    And I enter "192.168.5.254" in IP field
+    And I click on "Save Formula"
+    Then I should see a "Formula saved!" text
+
+@proxy
+@private_net
+  Scenario: Parametrize DHCP on the branch server
+    Given I am on the Systems overview page of this "proxy"
+    When I follow "Formulas" in the content area
+    And I follow first "Dhcpd" in the content area
+    And I enter "example.org" in domain name field
+    And I enter "192.168.5.254" in domain name server field
+    And I enter "eth1" in listen interfaces field
+    And I enter "192.168.5.0" in network IP field
```

Comment:
Idea: If we will use those IPs in other test cases or steps, it would be awesome to have them in some properties file or similar and use variables, so if there is a change on one of them, we don't need to find/replace in the whole doc.

Reply:
The retail network's IPs (`192.168.5.0/24`, with the proxy on `.254`) are completely hardcoded in Retail tests. I know we are taught to always use variable / constants, but in this testing context I don't see any advantage, well on the contrary. I find it a lot more readable and understandable this way. I don't see at all why it should change, and even if it happened it would be easy to fix.

Remember this is the readable, English-like description and it *is* a description by itself. We use variables in the steps.

Reply:
Okey, you are right, using properties/variables is less natural. It looks harder to maintain if an IP change, but I buy it :)